client/go/outline/vpn/device.go

```diff
@@ -93,7 +90,10 @@ func (d *RemoteDevice) RefreshConnectivity(ctx context.Context) (err error) {
 	}
 
 	slog.Debug("remote device is testing connectivity of server...")
-	tcpErr, udpErr := connectivity.CheckTCPAndUDPConnectivity(d.sd, d.pl)
+	// We need a PacketListener for the connectivity check.
+	// We can create one from the PacketProxy.
+	listener := &config.PacketProxyListener{Proxy: d.remote}
+	tcpErr, udpErr := connectivity.CheckTCPAndUDPConnectivity(d.sd, listener)
```

Comment:
Something like this
```suggestion
    udpErr := go pp.Start()
	tcpErr := connectivity.CheckTCPConnectivity(d.sd)
```

Reply:
But we have a start method in Start. So we need to define a new vpn.PacketProxy with Start that does whatever test and selection we need. (we will also need a NotifyNetworkChange to update the UDP support)